src/_types.ts

```diff
@@ -1,11 +1,23 @@
 import { Pointer } from "./pointers.ts";
-import { f32, f64, i16, i32, i64, i8, u16, u32, u64, u8 } from "./types.ts";
+import {
+  AllocatableStruct,
+  AllocatableStructConstructor,
+  f32,
+  f64,
+  i16,
+  i32,
+  i64,
+  i8,
+  TypedArray,
+  u16,
+  u32,
+  u64,
+  u8,
+} from "./types.ts";
 import { DynamicLibrary, DynamicLibraryInterface } from "./_library.ts";
 
-declare const _: unique symbol;
-
-export type PlatformPointer<T> = { [_]: "PlatformPointer" };
-export type PlatformString = { [_]: "PlatformString" };
+export type PlatformPointer<T> = Deno.PointerValue;
+export type PlatformString = Deno.PointerValue;
```

Comment:
Again this isn't a change I'm really interested in. I've put a lot of work into abstracting away the underlying platform and here you are putting it back in.

Reply:
but without this change, deno will complain irrelevant type at `src/deno/_pointer.ts`

Reply:
Are intended type conversion necessary? Or we can give actual pointer type from `Platform` interface

Reply:
What type `PlatformPointer` or `PlatformString` actually is should be hidden away from the end user. See [denoToPlatformPointer](https://github.com/smack0007/SDL_ts/blob/main/src/deno/_pointers.ts#L6). The function is actually returning a `Deno.UnsafePointer` but the end user doesn't know that and shouldn't have to know that. There is no actual type conversion happening, we are just telling TypeScript to pretend like it's some generic type.